api/clients/codecs/blob_codec.go

```diff
@@ -4,22 +4,22 @@ import (
 	"fmt"
 )
 
-type BlobEncodingVersion byte
+type PayloadEncodingVersion byte
 
 const (
-	// This minimal blob encoding contains a 32 byte header = [0x00, version byte, uint32 len of data, 0x00, 0x00,...]
+	// PayloadEncodingVersion0 entails a 32 byte header = [0x00, version byte, big-endian uint32 len of payload, 0x00, 0x00,...]
 	// followed by the encoded data [0x00, 31 bytes of data, 0x00, 31 bytes of data,...]
-	DefaultBlobEncoding BlobEncodingVersion = 0x0
+	PayloadEncodingVersion0 PayloadEncodingVersion = 0x0
 )
 
 type BlobCodec interface {
 	DecodeBlob(encodedData []byte) ([]byte, error)
 	EncodeBlob(rawData []byte) ([]byte, error)
 }
 
-func BlobEncodingVersionToCodec(version BlobEncodingVersion) (BlobCodec, error) {
+func BlobEncodingVersionToCodec(version PayloadEncodingVersion) (BlobCodec, error) {
 	switch version {
-	case DefaultBlobEncoding:
+	case PayloadEncodingVersion0:
 		return DefaultBlobCodec{}, nil
```

Comment:
this looks a bit weird now. Why is payload version 0 mapping to a defaultBlobCodec? Should we use BlobCodecVersion0 instead? Or do we want separate versioning for the payloadEncoding and BlobCodec? Or are these things the same?

Reply:
I renamed the constant to what I felt made sense for the new code, and but the name doesn't work so well in the old `blob_codec` code. If I *don't* rename the constant, then the name doesn't work so well in the new code

> Or are these things the same?

Yes, they are essentially the same thing.

Reply:
I see! Think I'm missing the venn diagram of which code is used for v1, which is used for v2, and which is used for both. is it not possible to completely separate v1 and v2 stuff?

Reply:
[Mostly](https://github.com/Layr-Labs/eigenda/pull/1120#discussion_r1961810221) separated in [3db92747](https://github.com/Layr-Labs/eigenda/pull/1120/commits/3db927475247142dbcd7e715a04cc5d199ea1fde). Hopefully this makes things a bit clearer

Reply:
resolving in favor of that other thread. Thanks for the change!